src/Raven.Client/Documents/Session/AsyncDocumentSession.ConditionalLoad.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Commands;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Session
+{
+    public partial class AsyncDocumentSession
+    {
+        /// <inheritdoc />
+        public async Task<(T Entity, string ChangeVector)> ConditionalLoadAsync<T>(string id, string changeVector, CancellationToken token = default)
+        {
+            using (AsyncTaskHolder())
+            {
+                if (string.IsNullOrEmpty(id))
+                    throw new ArgumentNullException(nameof(id));
+
+                if (Advanced.IsLoaded(id))
+                {
+                    var e = await LoadAsync<T>(id, token).ConfigureAwait(false);
+                    var cv = Advanced.GetChangeVectorFor(e);
+                    return (e, cv);
+                }
+
+                if (string.IsNullOrEmpty(changeVector))
+                    throw new InvalidOperationException($"The requested document with id '{id} is not loaded into the session and could not conditional load when {nameof(changeVector)} is null or empty.");
+
+                IncrementRequestCount();
+                var cmd = new ConditionalGetDocumentsCommand(id, changeVector);
+                await Advanced.RequestExecutor.ExecuteAsync(cmd, Advanced.Context, sessionInfo: null, token).ConfigureAwait(false);
+
+                switch (cmd.StatusCode)
+                {
+                    case HttpStatusCode.NotModified:
+                        return (default, changeVector); // value not changed
+                    case HttpStatusCode.NotFound:
+                        return default; // value is missing
```

Comment:
Should this be added to `TrackEntity` as we do for `Load`?
@ayende

Reply:
I think so, we now know it doesn't exist. 
It should be added to `missing`, no?

Reply:
sending 'Load' operation 2 times with the same non-existent ID will send 2 requests to the server, therefore it is not added to `TrackEntity` if the the entity doesn't exists on server

Reply:
Why do you think the 2nd call of `ConditionalLoadAsync` with the same non-existent ID should sent the request to the server. We already know that it doesn't exist, don't we? So we should answer immediately without asking the server.

Reply:
Yes, it should update `_knownMissingIds` and for that matter, the `ConditionalLoad` should also check it.

Reply:
`_knownMissingIds` missing ids is used in `Delete` and `Include`.
If you `Load` and entity with the same id twice (non-existing one) it will generate 2 requests.

Reply:
That isn't how it works. `LoadOperation` calls to `CheckIfIdAlreadyIncluded` which handles this.
No two calls here.

Reply:
Oh, we aren't _registering_ in the known missing in `Load`, yes, we need that too.

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-15615

I will add it to the ConditionalLoad as well